admin/gv_mail.php

```diff
@@ -4,421 +4,382 @@
-      error = true;
-    }
-  }
-//}
-function check_amount(field_name, field_size, message) {
-  if (form.elements[field_name] && (form.elements[field_name].type != "hidden")) {
-    var field_value = form.elements[field_name].value;
-
-    if (field_value == '' || field_value == 0 || field_value < 0 || field_value.length < field_size ) {
-      error_message = error_message + "* " + message + "\n";
-      error = true;
-    }
-  }
-}
-function check_message(msg) {
-  if (form.elements['message'] && form.elements['message_html']) {
-    var field_value1 = form.elements['message'].value;
-    var field_value2 = form.elements['message_html'].value;
-
-    if ((field_value1 == '' || field_value1.length < 3) && (field_value2 == '' || field_value2.length < 3)) {
-      error_message = error_message + "* " + msg + "\n";
-      error = true;
-    }
-  }
-}
-function check_input(field_name, field_size, message) {
-  if (form.elements[field_name] && (form.elements[field_name].type != "hidden")) {
-    var field_value = form.elements[field_name].value;
+            if ((field_value_input === '' || field_value_input.length < 1) && field_value_cust === '') {
+                error_message = error_message + "* " + message + "\n";
+                error = true;
+            }
+        }
 
-    if (field_value == '' || field_value.length < field_size) {
-      error_message = error_message + "* " + message + "\n";
-      error = true;
-    }
-  }
-}
+        //}
 
-function check_form(form_name) {
-  if (submitted == true) {
-    alert("<?php echo JS_ERROR_SUBMITTED; ?>");
-    return false;
-  }
-  error = false;
-  form = form_name;
-  error_message = "<?php echo JS_ERROR; ?>";
-
-  check_recipient('customers_email_address', 'email_to', "<?php echo ERROR_NO_CUSTOMER_SELECTED; ?>");
-  check_message("<?php echo ENTRY_NOTHING_TO_SEND; ?>");
-  check_amount('amount',1,"<?php echo ERROR_NO_AMOUNT_SELECTED; ?>");
-  check_input('subject','',"<?php echo ERROR_NO_SUBJECT; ?>");
-
-  if (error == true) {
-    alert(error_message);
-    return false;
-  } else {
-    submitted = true;
-    return true;
-  }
-}
-</script>
-<?php if ($editor_handler != '') include ($editor_handler); ?>
+        function check_input(field_name, field_size, message) {
+            if (form.elements[field_name] && (form.elements[field_name].type !== "hidden")) {
+                let field_value = form.elements[field_name].value;
+
+                if (field_value === '' || field_value.length < field_size) {
+                    error_message = error_message + "* " + message + "\n";
+                    error = true;
+                }
+            }
+        }
+
+        function check_amount(field_name, field_size, message) {
+            if (form.elements[field_name] && (form.elements[field_name].type !== "hidden")) {
+                let field_value = form.elements[field_name].value;
+
+                let decimal_check = /^[-+]?[0-9]+\.[0-9]+$/;

```

Comment:
Since `$currencies` is defined and in play, is there a reason not to use something like `$currencies->currencies[$_SESSION['currency']]['decimal_point']` as the divider instead of hard-coding a period? There are currencies that use a period for the thousands separator and a comma for the "decimal": 7.000,50 to be seven thousand and fifty parts of the currency. Is there some other constraint for which only a decimal is being permitted?